torchchat/distributed/checkpoint_utils.py

```diff
@@ -287,14 +260,15 @@ def update_state_dict(
         checkpoint_tensor = checkpoint[old_param]
         model_tensor = state_dict[param]
 
-        if "wq" in param:
-            checkpoint_tensor = permute_weight_to_attn_heads(
-                checkpoint_tensor, num_heads, head_dim, dim
-            )
-        elif "wk" in param:
-            checkpoint_tensor = permute_weight_to_attn_heads(
-                checkpoint_tensor, num_local_heads, head_dim, dim
-            )
+        if new_to_old_keymap is not None:
+            if "wq" in param:
```

Comment:
so this was the issue!  It seems like a flag as to the type of checkpoint is more robust and then map type to need for permuting?  Seems brittle if we have future checkpoints and we permute or not solely based on a fqn (i.e. generically, having 'wq' in a name doesn't actually imply anything inherently about need to permute)?

Reply:
I am using `if new_to_old_keymap is not None` as the flag for now.
If we are using original HF checkpoints, then we will need this `new_to_old_keymap`. Otherwise, if we use TC checkpoints, we don't.